patches/minecraft/net/minecraft/server/players/PlayerList.java.patch

```diff
@@ -138,14 +141,18 @@
     }
  
     @Nullable
-@@ -786,6 +_,7 @@
+@@ -786,9 +_,11 @@
        }
  
        this.m_11268_(new ClientboundUpdateTagsPacket(this.f_11195_.m_129895_().m_144442_(this.f_11206_)));
 +      net.minecraftforge.fmllegacy.network.NetworkHooks.syncCustomTagTypes(this.f_11195_.m_129895_());
        ClientboundUpdateRecipesPacket clientboundupdaterecipespacket = new ClientboundUpdateRecipesPacket(this.f_11195_.m_129894_().m_44051_());
  
        for(ServerPlayer serverplayer : this.f_11196_) {
++         net.minecraftforge.common.MinecraftForge.EVENT_BUS.post(new net.minecraftforge.event.SendDatapacksToClientEvent(serverplayer));
```

Comment:
So, two things. One are we wanting to garente this happens before or after tag sync? The two hooks make it happen at different times. I would vote before, I would also say that instead of firing this for every player on the server. I'd just make player @Nullable and pass in the PlayerList as well. So if the player == null send to all.
This will allow consumers to do optimizations like the one in this function where it shares the same packet across players. Also the name, it should be more 'OnDataConfigSentToClient" or something less stupid then what I just said. It's a generic 'this just happened' event not a 'you should do this thing.' event.